v2/reader.go

```diff
@@ -54,10 +70,29 @@ func (r *Reader) readHeader() (err error) {
 
 // CarV1ReaderAt provides an io.ReaderAt containing the CAR v1 dump encapsulated in this CAR v2.
 func (r *Reader) CarV1ReaderAt() io.ReaderAt {
+	return r.carv1SectionReader()
+}
+
+func (r *Reader) carv1SectionReader() *io.SectionReader {
 	return io.NewSectionReader(r.r, int64(r.Header.CarV1Offset), int64(r.Header.CarV1Size))
 }
 
 // IndexReaderAt provides an io.ReaderAt containing the carbs.Index of this CAR v2.
 func (r *Reader) IndexReaderAt() io.ReaderAt {
 	return internalio.NewOffsetReader(r.r, int64(r.Header.IndexOffset))
 }
+
+// ReadPragma reads the pragma from r.
+// This function accepts both CAR v1 and v2 payloads.
+// The roots are returned only if the version of pragma equals 1, otherwise returns nil as roots.
+func ReadPragma(r io.Reader) (version uint64, roots []cid.Cid, err error) {
+	header, err := carv1.ReadHeader(bufio.NewReader(r))
```

Comment:
A TODO for you: if I provide an io.Reader that already has all the fancy reader methods needed, you should use it directly rather than wrapping the Read method with something else like a bufio.Reader.

In this particular case, if I give you a bufio.Reader, you should use it directly. I imagine there are other cases like this in the carv2 API.

Reply:
I wish we could fix this in root carv1 by adding extra funcs but hey.